### ISSUE TITLE
Vacuum not handling YAML anchors correctly

### ISSUE THREAD

**Issue by TristanSpeakEasy** [daveshanley/vacuum#508]
Consider the two attached OpenAPI docs. One is a yaml document with anchors and the other is the same document that I have just inlined the anchors by doing a roundtrip through https://github.com/go-yaml/yaml

The below reproducible code shows that the document with anchors returns incorrect yaml nodes when present (specifically I am getting a node with no content).

```go
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/daveshanley/vacuum/model"
	"github.com/daveshanley/vacuum/motor"
	"github.com/daveshanley/vacuum/rulesets"
	"github.com/pb33f/libopenapi"
	"github.com/pb33f/libopenapi/datamodel"
	libopenapiUtils "github.com/pb33f/libopenapi/utils"
	"gopkg.in/yaml.v3"
)

type testRule struct{}

func (r *testRule) GetSchema() model.RuleFunctionSchema {
	return model.RuleFunctionSchema{
		Name: "test",
	}
}

func (r *testRule) RunRule(nodes []*yaml.Node,
	context model.RuleFunctionContext,
) []model.RuleFunctionResult {
	results := []model.RuleFunctionResult{}

	for _, schema := range context.Index.GetAllSchemas() {
		for _, nodeType := range []string{"anyOf", "allOf", "oneOf"} {
			keyNode, valNode := libopenapiUtils.FindKeyNode(nodeType, schema.Node.Content)
			if valNode == nil {
				continue
			}

			if len(valNode.Content) == 0 {
				results = append(results, model.RuleFunctionResult{
					Message:      "empty keyword found",
					Path:         schema.Path + "." + nodeType,
					RuleId:       context.Rule.Id,
					StartNode:    keyNode,
					EndNode:      valNode,
					RuleSeverity: context.Rule.Severity,
				})
			}
		}
	}

	return results
}

func (r *testRule) GetCategory() string {
	return "validation"
}

func main() {
	docConf := &datamodel.DocumentConfiguration{
		AllowRemoteReferences:               true,
		AllowFileReferences:                 true,
		IgnorePolymorphicCircularReferences: true,
		IgnoreArrayCircularReferences:       true,
		ExtractRefsSequentially:             true,
	}

	// data, err := os.ReadFile("openapi.yaml")
	data, err := os.ReadFile("openapi-inlined.yaml")
	if err != nil {
		log.Fatalf("error: %v", err)
	}

	d, err := libopenapi.NewDocumentWithConfiguration(data, docConf)
	if err != nil {
		log.Fatalf("error: %v", err)
	}

	ex := &motor.RuleSetExecution{
		RuleSet: &rulesets.RuleSet{
			Rules: map[string]*model.Rule{
				"test": {
					Name:         "test",
					Id:           "test",
					Given:        "$",
					Resolved:     false,
					Severity:     "error",
					Formats:      model.OAS3AllFormat,
					RuleCategory: model.RuleCategories[model.CategoryValidation],
					Type:         rulesets.Validation,
					Then: model.RuleAction{
						Function: "test",
					},
				},
			},
		},
		Document: d,
		PanicFunction: func(err any) {
			log.Fatalf("error: %v", err)
		},
		CustomFunctions: map[string]model.RuleFunction{
			"test": &testRule{},
		},
		AllowLookup: true,
	}

	results := motor.ApplyRulesToRuleSet(ex)

	for _, result := range results.Results {
		log.Printf("result: %v", result)
	}

	if len(results.Results) > 0 {
		log.Fatalf("errors found")
	}

	fmt.Println("done")
}
```

The code returns `errors found` in the version with anchors and just returns `done` in the inlined version

[openapi.zip](https://github.com/user-attachments/files/15646101/openapi.zip)

**Comment by Jakousa** [comment]
This bug makes vacuum unusable for any real world use cases. Here's another example:

Given file:
```
openapi: 3.0.3

x-a: &anchor
  important-field: true

x-b:
  <<: *anchor
```
and rule set
```
rules:
  test-rule:
    description: Important field must be defined on all levels
    message: "important-field must be defined"
    severity: error
    given: $.x-b
    then:
      field: important-field
      function: defined
```

Spectral works as expected
```
$ spectral lint config.yaml --ruleset ruleset.yaml
No results with a severity of 'error' found!
```

Vacuum does not:
```
$ vacuum -r ruleset.yaml lint routes-openapi-proposal-1.yaml -d
Location                           | Severity | Message                         | Rule      | Category   | Path
routes-openapi-proposal-1.yaml:7:3 | error    | important-field must be defined | test-rule | Validation | $.x-b
```

**Comment by daveshanley** [comment]
> This bug makes vacuum unusable for any real world use cases. Here's another example:

Not quite sure how to respond to this. The 1m+ downloads tells me otherwise.

**Comment by Jakousa** [comment]
> > This bug makes vacuum unusable for any real world use cases. Here's another example:
> 
> Not quite sure how to respond to this. The 1m+ downloads tells me otherwise.

Sorry about the choice of words and the tone it left 🙇 I should've said "for our use cases". We were attempting to migrate to vacuum for its superior performance but ran into the issue with the anchors on most of our files. Overall the project is very promising, excited to retry it later ❤️

**Comment by daveshanley** [comment]
The issue has been fixed with the latest release of `libopenapi` all merge nodes, anchors and aliases are handled correctly. `vacuum` and `libopenapi` will be updated soon.

**Comment by daveshanley** [comment]
FYI: https://github.com/pb33f/libopenapi/commit/d1d998a79d8aa63b9aef9c6e9a8e96fe802c9e21

**Comment by daveshanley** [comment]
This has been fixed in `v0.11.1`

Anchors, Aliases and Merge Nodes are supported in lower level lookups in `libopenapi` now. 

@Jakousa using your same example, the following output is now rendered:

![Screenshot 2024-07-09 at 2 27 23 PM](https://github.com/daveshanley/vacuum/assets/187345/196808f2-ea0a-4ce6-ba11-2e135a57de9d)
